### ISSUE TITLE
[SDK] Make the Android Matrix SDK useable for Kotlin without Android

### ISSUE THREAD

**Issue by bluefireoly** [vector-im/element-android#3236]
As in theory a Matrix SDK does not require Android, it would be great if the Android Matrix SDK would not require Android as well.

This change would be great, because it would open up the ability to use it anywhere else with Kotlin.

People use Kotlin for much more than just Android, this is why a Matrix SDK for the language in general would make sense. I see great potential there, because that SDK would be a solid choice for all kinds of projects, due to the fact that it is backed by Element.

**Comment by bmarty** [comment]
I understand, but this is a Android Matrix SDK, so it relies on the Android framework so far.

**Comment by bluefireoly** [comment]
Could you give an assessment of how heavily this SDK is dependent on Android libraries?

**Comment by Dominaezzz** [comment]
Would basically need a full rewrite.
All the state is stored in Realm for starters, which is Android and iOS only iirc.
There's a whole lot of `LiveData` (which could be turned into `Flow`).
There's also some android lifecycle stuff lurking around and a fair bit of "main thread" safety code.
It's just not practical for the Element devs to make this work without a full rewrite.
Considering that they just did a full rewrite, I don't think they'll be doing one again anytime soon.

**Comment by bluefireoly** [comment]
Yeah I realized that too by having a deeper look, that's why I am now using your [matrix-kt](https://github.com/Dominaezzz/matrix-kt) library which works fine for me :)

Thank you @Dominaezzz <3
